bin/utils/subkey/src/main.rs

```diff
@@ -236,15 +236,37 @@ fn main() {
 	let matches = get_app(&usage).get_matches();
 
 	if matches.is_present("ed25519") {
-		return execute::<Ed25519>(matches)
+		return execute_proc::<Ed25519>(matches);
 	}
 	if matches.is_present("secp256k1") {
-		return execute::<Ecdsa>(matches)
+		return execute_proc::<Ecdsa>(matches)
 	}
-	return execute::<Sr25519>(matches)
+	return execute_proc::<Sr25519>(matches)
 }
 
-fn execute<C: Crypto>(matches: ArgMatches)
+#[derive(Debug, derive_more::Display, derive_more::From)]
+enum Error {
+	Static(&'static str),
+	Io(std::io::Error),
+	Formatted(String),
+}
+
+fn execute_proc<C: Crypto>(matches: ArgMatches)
```

Comment:
Just use `fn main() -> Result<(), Error>`

Reply:
It does not really pretty-print error this way, see `nv-sk-err-msin` branch

Reply:
If you do this: https://github.com/paritytech/substrate/blob/c3a4c55cfc3a5d6f236dc184ea76f5b9ebe1eae4/client/network/src/error.rs#L51

The error message should look good.

Reply:
Better, but still worse than here if you ask me (updated `nv-sk-err-msin` branch)

Reply:
The only different I see is that there is no new line after `Error:` in the output? I think hat is okay and less code is always better, IMHO^^

Reply:
@bkchr updated